File: spaceship_crm/java8/spaceship_crm_java8_jsp_jpa/src/main/java/com/spaceship/crm/entity/UserActivity.java
/**
 * Author veni.vidi.dev (veni.vidi.dev@gmail.com)
 */
package com.spaceship.crm.entity;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.CreatedBy;
import org.springframework.data.annotation.CreatedDate;

import javax.persistence.*;
import java.io.Serializable;
import java.util.Date;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
@IdClass(UserActivityId.class)
public class UserActivity {

    @Id
    @ManyToOne
    private User user;

    @Id
    @ManyToOne
    private Activity activity;

    @CreatedDate
    @Column(updatable = false)
    private Date createdAt;

    @CreatedBy
    @Column(updatable = false)
    private Long createdBy;

}

File: spaceship_crm/kotlin/spaceship_crm_kotlin_spring_mustache_jpa/src/main/kotlin/com/spaceship/crm/entity/UserActivity.kt
/**
 * Author veni.vidi.dev (veni.vidi.dev@gmail.com)
 */
package com.spaceship.crm.entity

import org.springframework.data.annotation.CreatedBy
import org.springframework.data.annotation.CreatedDate
import java.io.Serializable
import java.util.*
import javax.persistence.*

@Entity
@IdClass(UserActivityId::class)
class UserActivity(
        @Id
        @ManyToOne
        val user: User,

        @Id
        @ManyToOne
        val activity: Activity,

        @CreatedDate
        @Column(updatable = false)
        val createdAt: Date,

        @CreatedBy
        @Column(updatable = false)
        val createdBy: Long? = null
)

class UserActivityId(
        var user: Long? = null,
        var activity: Long? = null
) : Serializable